opentelemetry/api/trace/__init__.py

```diff
@@ -0,0 +1,190 @@
+        current span exists. Start the span and set it as the current span in
+        this tracer's context.
+
+        On exiting the context manager stop the span and set its parent as the
+        current span.
+
+        Example::
+
+            with tracer.span("one") as parent:
+                parent.add_event("parent's event")
+                with tracer.span("two") as child:
+                    child.add_event("child's event")
+                    tracer.get_current_span()  # returns child
+                tracer.get_current_span()      # returns parent
+            tracer.get_current_span()          # returns the previously active span
+
+        Args:
+            name: The name of the span to be created.
+            parent: This span's parent.
+
+        Raises:
+            ValueError: if ``name`` is null.
+        """
+        raise NotImplementedError
+
+    def create_span(self, name: str, parent: Span=None) -> Span:
+        """Create a new span as a child of the currently active span.
+
+        If called with ``parent`` this method won't affect the tracer's
+        context.
+
+        See ``span`` for a context manager that controls the span's lifetime.
+
+        Args:
+            name: The name of the span to be created.
+            parent: This span's parent.
+
+        Raises:
+            ValueError: if ``name`` is null.
+        """
+        raise NotImplementedError
+
+
+class Span(object):
+    """A span represents a single operation within a trace.
+    """
+
+    def start(self) -> None:
+        """Set the current time as the span's start time.
+
+        Each span represents a single operation. The span's start time is the
+        wall time at which the operation started.
+
+        Only the first call to ``start`` should modify the span, and
+        implementations are free to ignore or raise on further calls.
+        """
+        raise NotImplementedError
+
+    def end(self) -> None:
+        """Set the current time as the span's end time.
+
+        The span's end time is the wall time at which the operation finished.
+
+        Only the first call to ``end`` should modify the span, and
+        implementations are free to ignore or raise on further calls.
+        """
+        raise NotImplementedError
+
+    def get_context(self) -> SpanContext:
+        """Get an immutable snapshot of this span.
+
+        Returns:
+            A :class:`.SpanContext` with a copy of this span's immutable state.
+        """
+        raise NotImplementedError
+
+
+class SpanContext(object):
+    """The state of a Span to propagate between processes.
+
+    This class includes the immutable attributes of a :class:`.Span` that must
+    be propagated to a span's children and across process boundaries.
+
+    Args:
+        trace_id: The ID of the trace that this span belongs to.
+        span_id: This span's ID.
+        options: Global trace options to propagate.
```

Comment:
Curious, why would we want to emphasize "global"? It doesn't seem to be a global thing.

Reply:
+1

Reply:
I think its global for the Span's trace, as opposed to affecting that span only.

Reply:
It is possible that spans under the same trace have different tracestate.

Reply:
I took "global" from the `TraceOptions` java docs: https://github.com/c24t/opentelemetry-java/blob/8b1865303ae46ff419a7dc803fcd4e2552b45314/api/src/main/java/io/opentelemetry/trace/TraceOptions.java#L25. I thought we'd change the state between spans but not the options.

In any case "global" seems wrong here, and we should consider changing the wording in java too.